docs/install/from_source.rst

```diff
@@ -0,0 +1,154 @@
+scratch on various systems. It consists of two steps:
+
+1. First build the shared library from the C++ codes (`libtvm.so` for linux/osx and `libtvm.dll` for windows).
+2. Setup for the language packages (e.g. Python Package).
+
+To get started, clone tvm repo from github. It is important to clone the submodules along, with ``--recursive`` option.
+
+.. code:: bash
+
+    git clone --recursive https://github.com/dmlc/tvm
+
+For windows users who use github tools, you can open the git shell, and type the following command.
+
+.. code:: bash
+
+   git submodule init
+   git submodule update
+
+
+Build the Shared Library
+------------------------
+
+Our goal is to build the shared libraries:
+
+- On Linux the target library are `libtvm.so, libtvm_topi.so`
+- On OSX the target library are `libtvm.dylib, libtvm_topi.dylib`
+- On Windows the target library are `libtvm.dll, libtvm_topi.dll`
+
+
+.. code:: bash
+
+    sudo apt-get update
+    sudo apt-get install -y python python-dev python-setuptools gcc libtinfo-dev zlib1g-dev
+
+The minimal building requirements are
+
+- A recent c++ compiler supporting C++ 11 (g++-4.8 or higher)
+- CMake 3.5 or higher
+- We highly recommend to build with LLVM to enable all the features.
+- It is possible to build without llvm dependency if we only want to use CUDA/OpenCL
+
+We use cmake to build the library.
+The configuration of tvm can be modified by `config.cmake`.
+
+
+- First, check the cmake in your system, you do not have cmake
+  you can obtain the latest version from `official website <https://cmake.org/download/>`_
+- First create a build directory, copy the ``cmake/config.cmake`` to the directory.
+
+  .. code:: bash
+
+      mkdir build
+      cp cmake/config.cmake build
+
+- Edit ``build/config.cmake`` to customize the compilation options
+
+  - On macOS, for some versions of XCode, you need to add ``-lc++abi`` in the LDFLAGS or you'll get link errors.
+
+- TVM optionally depends on LLVM. LLVM is required for CPU codegen that needs LLVM.
+
+  - LLVM 4.0 or higher is needed for build with LLVM. Note that verison of LLVM from default apt may lower than 4.0.
+  - Since LLVM takes long time to build from source, you can download pre-built version of LLVM from
+    [LLVM Download Page](http://releases.llvm.org/download.html).
+
+
+    - Unzip to a certain location, modify ``build/config.cmake`` to add ``set(USE_LLVM /path/to/your/llvm/bin/llvm-config)``
+    - You can also directly set ``set(USE_LLVM ON)`` and let cmake search for a usable version of LLVM.
+
+  - You can also use [LLVM Nightly Ubuntu Build](https://apt.llvm.org/)
+
+    - Note that apt-package append ``llvm-config`` with version number.
+      For example, set ``set(LLVM_CONFIG llvm-config-4.0)`` if you installed 4.0 package
+
+- We can then build tvm and related libraries.
+
+  .. code:: bash
+
+      cd build
+      cmake ..
+      make -j4
+
+If everything goes well, we can go to the specific language installation section.
```

Comment:
Add reference/hyperlink

Reply:
sorry I merged this in before seeing your comment, can you send a PR to add the fix you mentioned?

Reply:
these are minor cosmetic changes to the doc - so I won't be PRing those.